Does not install properly with Yarn

### Steps to reproduce

**Tell us about your environment:**

* ndb version: v1.0.6, attempted v1.0.5, v1.0.4, and v1.0.3 as well
* Platform / OS version: MacOS High Sierra 10.13.3
* Node.js version: Attempted with both v8.9.4 and v10.7.0

**What steps will reproduce the problem?**

1. Run `ndb .` or `ndb <filename>.js`
2. `ndb` does indeed launch, but only shows a black window.

**What is the expected result?**

The proper debugger UI should show up.

**What happens instead?**

None of the UI renders.

<img width="1200" alt="screen shot 2018-07-22 at 9 00 23 pm" src="https://user-images.githubusercontent.com/8698364/43055175-55aa202a-8df2-11e8-9463-8c460c9f3a39.png">

I'm experiencing this on `MacOS High Sierra 10.13.4` and `node/10.7.0` and here's what I see in my console after running `ndb` (with and without arguments):

```
(node:28405) UnhandledPromiseRejectionWarning: Error: Cannot find module './.local-frontend/ndb_app.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.resolve (internal/modules/cjs/helpers.js:30:19)    at main (/usr/local/lib/node_modules/ndb/ndb.js:83:28)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

Also getting a blank screen on macOS, running it with `NDB_DEBUG_FRONTEND` gave me this:

```
$ NDB_DEBUG_FRONTEND=1 ndb .
Debugger listening on ws://127.0.0.1:57170/c22c9957-6780-4179-bee2-8bb00d3f6144
For help see https://nodejs.org/en/docs/inspector
(node:98311) UnhandledPromiseRejectionWarning: Error: Cannot find module 'chrome-devtools-frontend/front_end/node_modules/xterm/dist/xterm.css'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.resolve (internal/module.js:18:19)
    at requestIntercepted (/Users/jasonrushton/.config/yarn/global/node_modules/ndb/ndb.js:214:24)
    at emitOne (events.js:116:13)
    at Page.emit (events.js:211:7)
    at NetworkManager.Page._networkManager.on.event (/Users/jasonrushton/.config/yarn/global/node_modules/puppeteer/lib/Page.js:122:74)
    at emitOne (events.js:116:13)
    at NetworkManager.emit (events.js:211:7)
    at NetworkManager._handleRequestStart (/Users/jasonrushton/.config/yarn/global/node_modules/puppeteer/lib/NetworkManager.js:221:10)
    at NetworkManager._onRequestIntercepted (/Users/jasonrushton/.config/yarn/global/node_modules/puppeteer/lib/NetworkManager.js:169:12)
(node:98311) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:98311) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Debugger attached.```

--- a/scripts/build_release_application.js
+++ b/scripts/build_release_application.js
@@ -24,6 +24,15 @@ function minify(code) {
   }).code;
 }
 
+async function write(stream, data) {
+  let callback;
+  let promise = new Promise(resolve => callback = resolve);
+  const success = stream.write(data, 'utf8', callback);
+  if (!success)
+    await new Promise(resolve => stream.once('drain', resolve));
+  return promise;
+}
+
 class ReleaseBuilder {
   constructor(path, output) {
     this._path = path;
@@ -140,34 +149,37 @@ class ReleaseBuilder {
       input: fs.createReadStream(htmlFullName, {encoding: 'utf8'}),
       crlfDelay: Infinity
     });
+    const outputLines = [];
     rl.on('line', line => {
       if (line.includes('<script ') || line.includes('<link '))
         return;
       if (line.includes('</head>')) {
-        output.write(`    <script type="text/javascript" src="${app.name}.js"></script>\n`);
-        output.write(`    <script>${minify(jsContent)}</script>\n`);
+        outputLines.push(`    <script type="text/javascript" src="${app.name}.js"></script>\n`);
+        outputLines.push(`    <script>${minify(jsContent)}</script>\n`);
       }
-      output.write(`${line}\n`);
+      outputLines.push(`${line}\n`);
     });
     await new Promise(resolve => rl.on('close', resolve));
     rl.close();
+    for (const line of outputLines)
+      await write(output, line);
     output.close();
   }
 
   async _buildAppScript(app) {
     const scriptName = `${app.name}.js`;
     const output = fs.createWriteStream(path.join(this._output, scriptName));
-    output.write('/* Runtime.js */\n');
+    await write(output, '/* Runtime.js */\n');
     const runtimeFullName = this._lookupFile(`Runtime.js`);
-    output.write(minify(await fsReadFile(runtimeFullName, 'utf8')));
-    output.write(`allDescriptors.push(...${JSON.stringify(app.releaseModuleDescriptors)});`);
-    output.write(`/* Application descriptor ${app.name} */`);
-    output.write(`applicationDescriptor = ${JSON.stringify(app.descriptor)}`);
-    output.write('/* Autostart modules */;\n');
+    await write(output, minify(await fsReadFile(runtimeFullName, 'utf8')));
+    await write(output, `allDescriptors.push(...${JSON.stringify(app.releaseModuleDescriptors)});`);
+    await write(output, `/* Application descriptor ${app.name} */`);
+    await write(output, `applicationDescriptor = ${JSON.stringify(app.descriptor)}`);
+    await write(output, '/* Autostart modules */;\n');
     let content = (await Promise.all(app.autoStartModules.map(module => this._writeScripts(module)))).join('\n');
     content = minify(content);
-    output.write(content);
-    output.write(';\n/* Autostart resources */\n');
+    await write(output, content);
+    await write(output, ';\n/* Autostart resources */\n');
     const resources = [];
     for (const module of app.autoStartModules) {
       if (!module.resources) continue;
@@ -200,7 +212,7 @@ class ReleaseBuilder {
   async _writeResources(resources, output) {
     if (!resources)
       return;
-    output.write((await Promise.all(resources.map(async resource => {
+    await write(output, (await Promise.all(resources.map(async resource => {
       const fullName = this._lookupFile(resource);
       let content = await fsReadFile(fullName, 'utf8');
       content = content.replace(/\\/g, '\\\\');
@@ -217,7 +229,7 @@ class ReleaseBuilder {
         await util.promisify(fs.mkdir)(folder);
       const output = fs.createWriteStream(
           path.join(folder, `${module.name}_module.js`));
-      output.write(minify(await this._writeScripts(module)));
+      await write(output, minify(await this._writeScripts(module)));
       if (module.resources) {
         await this._writeResources(
             module.resources.map(resource => path.join(module.name, resource)), output);

